### DISCUSSION TITLE
How to apply a LineLoad on a line or at a boundary

### DISCUSSION THREAD

**Discussion by salaudeen-ya** [idaholab/moose#24669] [Q&A General] [answered]
Hello,

I know there is a way of applying a point load as a Dirac force in MOOSE. But is there a way of applying a line load that can or can not be applied on nodes?

**Comment by GiudGiud** [comment]
Hello

Which dimension is the mesh/system?
Is the line on the boundary or inside the domain?

Guillaume

**Reply by salaudeen-ya** [reply]
I need it for any dimension. I would be needing it for both inside the domain, and on a boundary

**Reply by GiudGiud** [reply]
So 2D and 3D?
For a boundary, then you can use boundary conditions to apply a load on a line in 2D

**Reply by salaudeen-ya** [reply]
Thank you.
How about inside a domain?

**Reply by GiudGiud** [reply]
good question
@cpgr @rpodgorney do you ever do that for porous flow?

**Reply by cpgr** [reply, marked as answer]
Can you make an internal 'boundary' using ExtraNodesetGenerator and then apply a BC internally? I've done that before.

**Reply by rpodgorney** [reply]
I’ve never used a line, tagging @lynnmunday as he may have other approaches for this

**Reply by lynnmunday** [reply]
You can specify a series of load points as coordinates in a reporter and then use the https://mooseframework.inl.gov/source/dirackernels/ReporterPointSource.html
You can read the points into the reporterPointSource from a csv file.

**Reply by jiangwen84** [reply]
Can you distribute the line load into a series of load points? If so, you can either use @lynnmunday 's Reporter approach, or directly provide a vector of points in the DiracKernel.
